File: Selenium/Project/CRM_Activity9.java
package project;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class CRM_Activity9 {
	
	//declare driver
		WebDriver driver;
		Actions action;
		WebDriverWait wait;

		
		@BeforeClass
		  public void beforeClass() {
			
			//Initializing
			  driver = new ChromeDriver();
			  action = new Actions(driver);
			  wait = new WebDriverWait(driver,Duration.ofSeconds(5));
			  
		  }
		
  @Test
  public void f() {
	  
	  	 //Login
		 CRM_Activity4 login = new CRM_Activity4();
		 login.verifyLogin(driver);
		 
		 //Navigate Sales->Leads
		 WebElement sales = driver.findElement(By.xpath("*//a[@id='grouptab_0']"));
		 action.moveToElement(sales).click().build().perform();
		 WebElement leads = driver.findElement(By.xpath("*//a[@id='grouptab_0']//following-sibling::ul[@class='dropdown-menu']/li/a[text()='Leads']"));
		 leads.click();
		 action.pause(5000L).build().perform();
		 
		//Printing names from table
		  for(int i=1; i<=10 ; i++)
		  {
					  
		  String nameText = driver.findElement(By.xpath("*//table[contains(@class,'table-responsive')]/tbody/tr["+i+"]/td[3]")).getText();
		  String userText = driver.findElement(By.xpath("*//table[contains(@class,'table-responsive')]/tbody/tr["+i+"]/td[8]")).getText();
		  
		  System.out.println(i+". Name : " +nameText +" ; User : "+userText);
		  
					  
		  }
  }
  
  @AfterClass
  public void afterClass() {
	  
	  driver.quit();
  }
  
}

File: Selenium/Project/CRM_Activity9.py
# Import 
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from CRM_Activity4 import verify_login
from selenium.webdriver import ActionChains

# Start the Driver
with webdriver.Chrome() as driver:
    # Navigate to the URL
        driver.get("https://alchemy.hguy.co/crm/")

        #Login
        verify_login(driver)

        #Navigate Sales -> Leads
        sales = driver.find_element(By.XPATH,"*//a[@id='grouptab_0']")
        actions = ActionChains(driver)
        actions.move_to_element(sales).click().perform()

        leads = driver.find_element(By.XPATH,"*//a[@id='grouptab_0']//following-sibling::ul[@class='dropdown-menu']/li/a[text()='Leads']")
        actions.move_to_element(leads).click().pause(5).perform()

        #Printing odd rows names from table
        for i in range(1, 11):
            name_text = driver.find_element(By.XPATH, f"//table[contains(@class,'table-responsive')]/tbody/tr[{i}]/td[3]").text
            user_text = driver.find_element(By.XPATH, f"//table[contains(@class,'table-responsive')]/tbody/tr[{i}]/td[8]").text
            print(f"{i}. Name: {name_text} ; User: {user_text}")

driver.quit()